src/emqx_schema_validate.erl

```diff
@@ -1,48 +1,80 @@
 -module(emqx_schema_validate).
 
 %% API exports
--export([main/1]).
+-export([main/1, do_spellcheck_schema/1]).
 
--define(TAB, stats_tab).
+-define(DEFAULT_MAX_CONCURRENCY, 10).
 
 %%====================================================================
 %% API functions
 %%====================================================================
 
-main(["-"]) ->
+main(Args) ->
+    main(Args, ?DEFAULT_MAX_CONCURRENCY).
+
+main(["-j", N0 | Rest], _MaxConcurrency) ->
+    N = list_to_integer(N0),
+    main(Rest, N);
+main(["-"], MaxConcurrency) ->
     Binary = iolist_to_binary(read_stdio()),
-    process_data(Binary);
-main([JsonFile]) ->
+    process_data(Binary, MaxConcurrency);
+main([JsonFile], MaxConcurrency) ->
     {ok, Binary} = file:read_file(JsonFile),
-    process_data(Binary);
-main(_) ->
+    process_data(Binary, MaxConcurrency);
+main(_, _) ->
     io:format("Usage: emqx_schema_validate <path-to-json-schema-dump>~n", []),
     erlang:halt(1).
 
 %%====================================================================
 %% Internal functions
 %%====================================================================
 
-process_data(Binary) ->
+process_data(Binary, MaxConcurrency) ->
     {ok, _} = application:ensure_all_started(emqx_schema_validate),
     langtool:start(),
     Data = jsone:decode(Binary, [{object_format, map}, {keys, atom}]),
-    spellcheck_schema(Data),
+    spellcheck_schema(Data, MaxConcurrency),
     case is_ok() of
         true -> halt(0);
         false -> halt(1)
     end.
 
-spellcheck_schema(Data) ->
+spellcheck_schema(Data, MaxConcurrency) ->
```

Comment:
This code is extremely complicated (untested, undocumented) for such a simple single-use script.
 
Use some standard function like `rpc:pmap`?

Reply:
`rpc:pmap` does not control concurrency.

Is it so extremely complicated, really?  :thinking: 

I tested manually with a few simple cases, as in the description.  Given there were no automated tests before...

Reply:
> rpc:pmap does not control concurrency

It should be possible to use e.g. `lists:sublist` to split the entire list into chunks and run `rpc:pmap()` on chunks. Assuming the run time of each request is about equal, it should be fine enought.

> Is it so extremely complicated, really? 🤔

Complicated enough that I wouldn't want to spend time debugging it if it fails in CI.

Reply:
> It should be possible to use e.g. lists:sublist to split the entire list into chunks and run rpc:pmap() on chunks. Assuming the run time of each request is about equal, it should be efficient enough.

It's basically the same thing?  After each process terminates, we spawn a single one with the next node.  Then we wait for all processes to die.

> Complicated enough that I wouldn't want to spend time debugging it if it fails in CI. Reasoning about message passing and processes is harder than just list processing.

There isn't any message passing here.  Since I didn't introduce any new logic really, only used existing logic in spawned processes, I'm still a bit puzzled about what's terrible about this approach.  :thinking:

Reply:
> It's basically the same thing? After each process terminates, we spawn a single one with the next node. Then we wait for all processes to die.

If it's basically the same, then it should be an easy fix?

> There isn't any message passing here. Since I didn't introduce any new logic really, only used existing logic in spawned processes, I'm still a bit puzzled about what's terrible about this approach. 🤔

Well, when I open schema spellcheck script I expect to only see code related to spellchecking. Not some task scheduling logic.